power_grid_model_c/power_grid_model/include/power_grid_model/main_model.hpp

```diff
@@ -251,6 +251,7 @@ class MainModelImpl<ExtraRetrievableTypes<ExtraRetrievableType...>, ComponentLis
         main_core::register_topology_components<GenericLoadGen>(state_, comp_topo);
         main_core::register_topology_components<GenericVoltageSensor>(state_, comp_topo);
         main_core::register_topology_components<GenericPowerSensor>(state_, comp_topo);
+        main_core::register_topology_components<Regulator>(state_, comp_topo);
```

Comment:
I did not use ComponentTopology in the ranking part as it does not have status too. Is there any added advantage to having this info here for outer loop?

Reply:
on my end, I need it because I need the U and I at the controlled node deducted immediately from the `math_output` to prevent needlessly extracting output of a gazillion components while I only need a couple every iteration.

On the input side, I don't particularly care either way. Probably, directly using `ID` lookup looks nicer but is more expensive. Since it only happens once either way, It's OK to keep it on that side, I guess